jax/_src/numpy/lax_numpy.py

```diff
@@ -6617,6 +6617,10 @@ def trace(a: ArrayLike, offset: int | ArrayLike = 0, axis1: int = 0, axis2: int
   util.check_arraylike("trace", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.trace is not supported.")
+
+  if axis1 == axis2:
+    raise ValueError("axis1 and axis2 can not be same")
```

Comment:
It's useful to print the axis values in the error message for better debuggability.

Reply:
Thanks! Included the axis values in the error message